source/base/kokkos.cc

```diff
@@ -31,9 +31,14 @@ namespace internal
   {
     if (!Kokkos::is_initialized())
       {
-        dealii_initialized_kokkos = true;
-        Kokkos::initialize();
-        std::atexit(Kokkos::finalize);
+        // only execute once
+        static bool dummy = [] {
+          dealii_initialized_kokkos = true;
+          Kokkos::initialize();
+          std::atexit(Kokkos::finalize);
+          return true;
+        }();
+        (void)dummy;
```

Comment:
Can we use `std::call_once` here?

Reply:
We could but I find it more cumbersome to use. If anyone requests it, I will change it.